### DISCUSSION TITLE
@auth_required return always flask_security.core.AnonymousUser

### DISCUSSION THREAD

**Discussion by unixshaman** [pallets-eco/flask-security#452] [Q&A] [answered]
Good day!

I use flask-security-too in my application [CerediraTess](https://github.com/Ceredira/CerediraTess/tree/fix_auth), branch fix_auth.When I run application, and login on Login page, i stay on Login page, cannot authorize.
I debug auth_required and see that user after correct password checking and saving in session, at the end of decorator again as flask_security.core.AnonymousUser.

Can help me? May be I do something wrong.

**Comment by jwag956** [comment, marked as answer]
You have a lot of code!
If you are staying on the login page then something is failing - have you logged at various points in your LoginForm? are you finding the user? And which of the following checks (in LoginForm) is failing? that might help narrow it down.
Also - try looking at the raw response HTML to see if there are errors that for some reason aren't being shown (i.e. something wrong with your template).

Also - I can't be sure - but it doesn't look like you are showing 'flashed' messages in your templates - which might cause confusion in other flask-security work flows.

**Reply by unixshaman** [reply]
Thanks for reply!
I have always return false here for every auth methods, after that code go to 396 as flask_security.core.AnonymousUser.
![image](https://user-images.githubusercontent.com/1523779/109938682-5fb33d00-7ce1-11eb-9f8a-329cd445ee57.png)

Is this the way it should be?

**Reply by unixshaman** [reply]
Thank you very much for the advice on finding the problem!

I created a new simple project exactly following the instructions from https://flask-security-too.readthedocs.io/en/stable/quickstart.html and found that everything works in the location shown in the screenshot. And I started looking for the difference between my project and example.
As a result, when I threw out all unnecessary things from my project and brought them exactly with an example, I got authorization.

P.S. Unfortunately, it was not possible to find out the reason for this behavior, since I changed most of the project entirely.

**Reply by unixshaman** [reply]
Working code in https://github.com/Ceredira/CerediraTess

**Reply by jwag956** [reply]
Looks like you made a lot of changes - so is it working correctly now.

The answer to 'should it always return False' is no - that means that login didn't validate the user.
